docs/tutorials/writing_a_new_data_struct.md

```diff
@@ -0,0 +1,104 @@
+# Writing a New Data Struct
+
+ICU4X is a heavily data-driven library. Most new features or components will require pulling in data from an external source.
+
+This tutorial aims to help ICU4X contributors add new data to the data pipeline. It is recommended that readers review [data_pipeline.md](../design/data_pipeline.md) for additional theory behind the design decisions in the data provider.
+
+## Lifecycle of ICU4X Data
+
+It is important to understand the phases of life of ICU4X data as it makes its way from the data source, like CLDR, to the data struct used at runtime. The following flowchart shows the phases and how they connect:
+
+![Lifecycle of Data in ICU4X](../assets/data_lifecycle.svg)
+
+The following steps take place at build time:
+
+1. First, the source data file is obtained from an external source. Examples could include the CLDR JSON release or the Unicode Character Database.
+2. Second, we use a Serde definition to deserialize the source data file. This Serde implementation does not need to be optimized for performance.
+3. Third, we transform from the source data struct to the ICU4X runtime data struct. This step can be expensive, because it is normally run as an offline build step.
+4. Fourth, the ICU4X runtime data struct is serialized to either JSON, if debugging is important, or to a blob store, if being prepared for use in production.
```

Comment:
Suggestion: It may be useful to explain the commands that perform step 2, 3, and 4 in our build pipeline so that contributor can have an clear example if they were to add a new ICU4X data structure. After reading this doc, I'm still unsure whether these steps are happening automagically during `cargo build` or not. Should I add something to the CI script, or should I manually call `icu4x-datagen` for a certain step?

Reply:
There is no `cargo build` magic (at least not yet) in the data build pipeline. You need to call `icu4x-datagen`.